src/accessors/IConfigurationExtend.ts

```diff
@@ -6,6 +8,12 @@ import { ISlashCommandsExtend } from './ISlashCommandsExtend';
  * of your Rocketlet. It is provided during initialization of your Rocketlet
  */
 export interface IConfigurationExtend {
+    /** Accessor for registering the assets which need to be. */
+    readonly assets: IAssetExtend;
+
+    /** Accessor for declaring the contextual bar buttons your Rocketlet provides. */
+    readonly contextualBarButtons: IContextualBarButtonsExtend;
```

Comment:
👍 I in fact meant `Icon` instead of `Button`. in ABAP:
```abap
REPLACE ALL OCCURENCES OF 'button` in above_comment WITH 'icon' IGNORING CASE.
```
However, also a button (icon+label+optional tooltip) could be created from the `IUiElementFactory`. For UI-Elements, believe constructor injection is fine. Thus, a button (a real button this time) and a ContextBarElement (in lack of a better name) could be created like that (consumer side):

```typescript
const myContainer = uiElementFactory.getContextBarElementFactory().createContextBarElement(
  uiElementFactory.getButtonFactory(). createButton(
    uiElementFactory.getIconFactory().createIcon('icon-github-circled'),
    uiTextFactory.createI18nText('Github'), //somehow solve internationalization
    uiTextFactory.createI18nText('Github-integration')
  ),
  uiAssetProvider.getAsset('main.html')
);
```